include/simdjson/numberparsing.h

```diff
@@ -569,20 +548,27 @@ static really_inline bool parse_number(const uint8_t *const buf, ParsedJson &pj,
       // we over-decrement by one when there is a '.'
       digit_count -= (start - start_digits);
       if (digit_count >= 19) {
-        // Ok, chances are good that we had an overflow!
-        // this is almost never going to get called!!!
-        // we start anew, going slowly!!!
-        return parse_float(buf, pj, offset, found_minus);
+        return parse_float_strtod(buf, pj, offset, p);
```

Comment:
i just figured it's so rare, maybe we're fine with this for imprecise parsing as well